src/Features/CSharp/Portable/CodeRefactorings/NamedArguments/CSharpAddNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,175 @@
+            }
+        }
+
+        protected override bool IsCandidate(SyntaxNode node)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.ElementAccessExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.BaseConstructorInitializer:
+                case SyntaxKind.ThisConstructorInitializer:
+                case SyntaxKind.ElementBindingExpression:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        protected override bool TryGetArguments(SyntaxNode node, SemanticModel semanticModel, out SeparatedSyntaxList<ArgumentSyntax> arguments)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                    {
+                        var syntaxNode = (InvocationExpressionSyntax)node;
+                        arguments = syntaxNode.ArgumentList.Arguments;
+                        break;
+                    }
+
+                case SyntaxKind.ElementAccessExpression:
+                    {
+                        var syntaxNode = (ElementAccessExpressionSyntax)node;
+
+                        // Array access is not allowed to contain named arguments
+                        var type = semanticModel.GetTypeInfo(syntaxNode.Expression).Type;
+                        if (type?.TypeKind == TypeKind.Array)
+                        {
+                            arguments = default(SeparatedSyntaxList<ArgumentSyntax>);
+                            return false;
+                        }
+
+                        arguments = syntaxNode.ArgumentList.Arguments;
+                        break;
+                    }
+
+                case SyntaxKind.ObjectCreationExpression:
+                    {
+                        var syntaxNode = (ObjectCreationExpressionSyntax)node;
+                        arguments = syntaxNode.ArgumentList.Arguments;
+                        break;
+                    }
+
+                case SyntaxKind.ElementBindingExpression:
+                    {
+                        var syntaxNode = (ElementBindingExpressionSyntax)node;
+                        arguments = syntaxNode.ArgumentList.Arguments;
+                        break;
+                    }
+
+                case SyntaxKind.BaseConstructorInitializer:
+                case SyntaxKind.ThisConstructorInitializer:
+                    {
+                        var syntaxNode = (ConstructorInitializerSyntax)node;
+                        arguments = syntaxNode.ArgumentList.Arguments;
+                        break;
+                    }
+
+                default:
+                    arguments = default(SeparatedSyntaxList<ArgumentSyntax>);
+                    return false;
+            }
+
+            return arguments.Count > 0 && arguments[0].NameColon == null;
```

Comment:
Why are we checking if  arguments[0].NameColon == null? Shouldn't this be handled in GetNamedArguments or am I missing something?